lib/util/napi.js

```diff
@@ -0,0 +1,156 @@
+"use strict";
+
+var fs = require('fs');
+var rm = require('rimraf');
+
+module.exports = exports;
+
+var versionArray = process.version
+	.substr(1)
+	.replace(/-.*$/, '')
+	.split('.')
+	.map(function(item) {
+		return +item;
+	});
+
+var napi_multiple_commands = [
+	'build',
+	'clean',
+	'configure',
+	'package',
+	'publish',
+	'reveal',
+	'testbinary',
+	'testpackage',
+	'unpublish'
+];
+
+var napi_build_version_tag = 'napi_build_version=';
+
+module.exports.get_napi_version = function() {
+	// returns the non-zero numeric napi version or undefined if napi is not supported.
+	var version = process.versions.napi; // can be undefined
+	if (!version) { // this code should never need to be updated
+		if (versionArray[0] === 9 && versionArray[1] >= 3) version = 2; // 9.3.0+
+		else if (versionArray[0] === 8) version = 1; // 8.0.0+
+	}
+	return version;
+};
+
+module.exports.get_napi_version_as_string = function() {
+	// returns the napi version as a string or an empty string if napi is not supported.
+	var version = module.exports.get_napi_version();
+	return version ? ''+version : '';
+};
+
+module.exports.validate_package_json = function(package_json) { // return err
+	var binary = package_json.binary;
+	var module_path_ok = binary.module_path && binary.module_path.includes('{napi_build_version}');
+	var remote_path_ok = binary.remote_path && binary.remote_path.includes('{napi_build_version}');
+	var package_name_ok = binary.package_name && binary.package_name.includes('{napi_build_version}');
+	var napi_build_versions = module.exports.get_napi_build_versions(package_json);
+
+	if (napi_build_versions) {
+		napi_build_versions.forEach(function(napi_build_version){
+			if (!((parseInt(napi_build_version) === napi_build_version) && (napi_build_version > 0))) {
```

Comment:
I'd be explicit about the base and write `parseInt(napi_build_version, 10)`.  Likewise on line 119.

Aside: your clauses have some superfluous parentheses.